### PR TITLE
Feature/dropdown working 2

### PR DESCRIPTION
**Pull request by IsaacHook**
see 57269ea6199c1487673fb6d53541463de0873235 for this PR changes

### CODE REVIEW

File: src/nodeUpdate.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { Schema, Kinds, Node, GetParams } from "./state.js"
+
+import { CanvasContext } from "./context.js"
+import { EditorAction, updateNode } from "./actions.js"
+
+export type NodeUpdateBehaviour = <S extends Schema>(node: Node<S>, action: string) => void
+
+export function makenodeUpdateBehaviour<S extends Schema>(
```
Comment (reecelondon) [inline src/nodeUpdate.ts]: camelCase ?

File: src/reduce.ts [hunk 2]
```diff
@@ -169,14 +172,14 @@ function createInitialNode<S extends Schema>(
 }
 
 
-const createNode = <S extends Schema>(kinds: Kinds<S>, state: EditorState<S>, action: EditorAction<S>):  EditorState<S> => {
-	const { id, kind, position } = action
+const createNode = <S extends Schema>(kinds: Kinds<S>, state: EditorState<S>, action: CreateNodeAction):  EditorState<S> => {
+	const { id, archetype, position } = action
 	const nodes = { ...state.nodes }
-	nodes[id] = createInitialNode(kinds, id, kind, position)
+	nodes[id] = createInitialNode(kinds, id, archetype, position, action.action)
```
Comment (alex-lucas) [inline src/reduce.ts]: Nit: Could be destructured above with new variable name

File: src/nodeUpdate.ts (hunk 1 above)
Comment (alex-lucas) [inline src/nodeUpdate.ts]: Nit: could be camelcase similar to `makeNodeDragBehavior`.

**Review by alex-lucas — COMMENTED** [review]
Small changes

**Review by alex-lucas — APPROVED** [review]

**Review by reecelondon — APPROVED** [review]

**Pull request merged**